python-avd/pyavd/_eos_designs/structured_config/underlay/router_bgp.py

```diff
@@ -25,53 +25,35 @@ def router_bgp(self: AvdStructuredConfigUnderlayProtocol) -> None:
         if not self.shared_utils.underlay_bgp:
             return
 
-        af_type = "ipv4" if not self.shared_utils.underlay_ipv6_numbered else "ipv6"
-
-        peer_group = EosCliConfigGen.RouterBgp.PeerGroupsItem(
-            name=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
-            type=af_type,
-            password=self.shared_utils.get_bgp_password(self.inputs.bgp_peer_groups.ipv4_underlay_peers),
-            bfd=self.inputs.bgp_peer_groups.ipv4_underlay_peers.bfd or None,
-            maximum_routes=12000,
-            send_community="all",
-        )
-        if self.inputs.bgp_peer_groups.ipv4_underlay_peers.structured_config:
-            self.custom_structured_configs.nested.router_bgp.peer_groups.obtain(self.inputs.bgp_peer_groups.ipv4_underlay_peers.name)._deepmerge(
-                self.inputs.bgp_peer_groups.ipv4_underlay_peers.structured_config, list_merge=self.custom_structured_configs.list_merge_strategy
-            )
-
-        if self.shared_utils.is_cv_pathfinder_router:
-            peer_group.route_map_in = "RM-BGP-UNDERLAY-PEERS-IN"
-            if self.shared_utils.wan_ha:
-                peer_group.route_map_out = "RM-BGP-UNDERLAY-PEERS-OUT"
-                if self.shared_utils.use_uplinks_for_wan_ha:
-                    # For HA need to add allowas_in 1
-                    peer_group.allowas_in._update(enabled=True, times=1)
-
-        self.structured_config.router_bgp.peer_groups.append(peer_group)
-
-        # Address Families
-        # TODO: - see if it makes sense to extract logic in method
-        if not self.shared_utils.underlay_ipv6_numbered:
-            address_family_ipv4_peer_group = EosCliConfigGen.RouterBgp.AddressFamilyIpv4.PeerGroupsItem(
-                name=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name, activate=True
-            )
-            if self.inputs.underlay_rfc5549 is True:
-                address_family_ipv4_peer_group.next_hop.address_family_ipv6._update(enabled=True, originate=True)
-
-            self.structured_config.router_bgp.address_family_ipv4.peer_groups.append(address_family_ipv4_peer_group)
-
-        if self.shared_utils.underlay_ipv6:
-            self.structured_config.router_bgp.address_family_ipv6.peer_groups.append_new(
-                name=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name, activate=True
-            )
+        # Set BGP peer group only when underlay link is present.
+        if self._underlay_p2p_links:
```

Comment:
It seems like you could rewrite this like this and exit early and deindent the rest of the function given that both line 55 and 79 then loop over the `underlay_p2p_links` so would do nothing if it is empty.

```suggestion
        if not self._underlay_p2p_links:
            return
        # <...> the rest of the function
```

Reply:
Done.